Sources/PackageGraph/VersionSetSpecifier.swift

```diff
@@ -225,8 +276,8 @@ extension VersionSetSpecifier {
             }
 
             if lhs.lowerBound == rhs {
-                // Return empty if the range represents the exact version.
-                if lhs.lowerBound == lhs.upperBound {
+                // Return empty if the range represents the exact version or the range is empty.
+                if lhs.lowerBound == lhs.upperBound || lhs.lowerBound.nextPatch() == lhs.upperBound {
```

Comment:
this change is fine, but I would probably split it into two if conditions with detailed comment above the special case we are adding / fixing now, since its non-trivial to understand from the code. ie


```
// Return empty if the range represents the exact version.
 if lhs.lowerBound == lhs.upperBound {
  return .empty
}
// detailed explanation 
 if lhs.lowerBound.nextPatch() == lhs.upperBound {
  return .empty
}
```

Reply:
Had automerging turned on, but will change in a follow-up

Reply:
https://github.com/apple/swift-package-manager/pull/5740